lib/orderbook/OrderBook.ts

```diff
@@ -141,15 +140,22 @@ class OrderBook extends EventEmitter {
   }
 
   private updateOrderQuantity = (type: { [pairId: string]: Orders }, order: orders.StampedOrder, decreasedQuantity: number) => {
-    const object = this.getOrderMap(type, order);
-    object[order.id].quantity = object[order.id].quantity - decreasedQuantity;
-    if (object[order.id].quantity === 0) {
-      delete object[order.id];
+    const identifier = this.getIdentifier(order);
+
+    const orderMap = this.getOrderMap(type, order);
+    orderMap[order[identifier]].quantity = orderMap[order[identifier]].quantity - decreasedQuantity;
+    if (orderMap[order[identifier]].quantity === 0) {
+      delete orderMap[order[identifier]];
     }
   }
 
   private addOrder = (type: { [pairId: string]: Orders }, order: orders.StampedOrder) => {
-    this.getOrderMap(type, order)[order.id] = order;
+    const identifier = this.getIdentifier(order);
+    this.getOrderMap(type, order)[order[identifier]] = order;
+  }
+
+  private getIdentifier = (order: orders.StampedOrder): string => {
```

Comment:
I'm unsure about this `getIdentifier` method. I think it's simpler to only use `id` as the key for referencing orders. Mainly because peers who try to fill our orders will not know the internal id, only the global id, and will reference orders using the global id, meaning we need to track some mapping of internal id to global id anyhow. So my preferred approach would be to maintain this mapping and do a lookup of id whenever we are given an id from the API layer. 

If we do decide to keep this, it should probably be `static` or moved out of the `Orderbook` class, maybe even in a utils class because we might need to use it elsewhere down the road.

Reply:
I completely forgot about that. Having a map with the `localId` as key and `id` as value makes most sense you are right.